### PR TITLE
More timing stats around uploading files #1309

### PR DESCRIPTION
**Pull request by mstriemer**
This is probably total overkill but here it is. I'm still not sure what's going wrong. It seems like we must be getting a server error or something.

Depending on how long these errors take maybe it's a replication lag issue and the user isn't getting pinned to master. Then they'd access a row that doesn't exist and get a 404.

I'm looking for 404s to this endpoint in the logs but nothing yet.

### CODE REVIEW

File: static/js/common/upload-addon.js [hunk 1]
```diff
@@ -382,7 +382,7 @@
                                 $upload_field.trigger("upload_finished", [file]);
                             }
                         });
-                    }, 1000);
+                    }, 5000);
```
Comment (mstriemer) [inline static/js/common/upload-addon.js]: I increased the timeout to reduce the chance we're reading while the server is writing but I don't see any evidence that this is an issue.

File: apps/devhub/tasks.py [hunk 2]
```diff
@@ -165,12 +165,13 @@ def validate_file(file_id, hash_, **kw):
 def handle_upload_validation_result(results, upload_pk, annotate=True):
     """Annotates a set of validation results, unless `annotate` is false, and
     saves them to the given FileUpload instance."""
-    if annotate:
-        results = annotate_validation_results(results)
+    with statsd.timer('devhub.validator.handle_upload_validation_result'):
```
Comment (kumar303) [inline apps/devhub/tasks.py]: We already have timing for each task. You can see this one at https://app.datadoghq.com/metric/explorer?live=true&page=0&is_auto=false&from_ts=1452625319843&to_ts=1452628919843&tile_size=m&exp_metric=addons_prod.tasks.devhub.tasks.handle_upload_validation_result.avg&exp_scope=&exp_agg=avg&exp_row_type=metric

File: apps/devhub/tasks.py [hunk 3]
```diff
@@ -181,11 +182,12 @@ def handle_upload_validation_result(results, upload_pk, annotate=True):
 def handle_file_validation_result(results, file_id, annotate=True):
     """Annotates a set of validation results, unless `annotate is false, and
     saves them to the given File instance."""
-    if annotate:
-        results = annotate_validation_results(results)
+    with statsd.timer('devhub.validator.handle_file_validation_result'):
```
Comment (kumar303) [inline apps/devhub/tasks.py]: this one exists at https://app.datadoghq.com/metric/explorer?live=true&page=0&is_auto=false&from_ts=1452542602739&to_ts=1452629002739&tile_size=m&exp_metric=addons_prod.tasks.devhub.tasks.handle_file_validation_result.avg&exp_scope=&exp_agg=avg&exp_row_type=metric

File: apps/devhub/views.py [hunk 4]
```diff
@@ -677,8 +679,15 @@ def standalone_upload_detail(request, uuid):
 @dev_required
 @json_view
 def upload_detail_for_addon(request, addon_id, addon, uuid):
-    upload = get_object_or_404(FileUpload, uuid=uuid)
-    return json_upload_detail(request, upload, addon_slug=addon.slug)
+    with statsd.timer('devhub.upload_detail_for_addon'):
```
Comment (kumar303) [inline apps/devhub/views.py]: Every view is timed already as well. This one is at https://app.datadoghq.com/metric/explorer?live=true&page=0&is_auto=false&from_ts=1452542602739&to_ts=1452629002739&tile_size=m&exp_metric=addons_prod.view.devhub.views.upload_detail_for_addon.GET.avg&exp_scope=&exp_agg=avg&exp_row_type=metric

File: apps/devhub/views.py [hunk 5]
```diff
@@ -677,8 +679,15 @@ def standalone_upload_detail(request, uuid):
...
+        try:
+            upload = get_object_or_404(FileUpload, uuid=uuid)
+            return json_upload_detail(request, upload, addon_slug=addon.slug)
+        except:
+            statsd.incr('devhub.upload_detail_for_addon_error')
```
Comment (kumar303) [inline apps/devhub/views.py]: This looks useful. Can you also add a counters for 1) every case, regardless of success/error and 2) a count of every success. That will allow us to make a graph of what percentage of every upload is a failure.

File: apps/devhub/views.py [hunk 6]
```diff
@@ -874,10 +883,20 @@ def upload_validation_context(request, upload, addon_slug=None, addon=None,
 
 @login_required
 def upload_detail(request, uuid, format='html'):
-    upload = get_object_or_404(FileUpload, uuid=uuid)
-
     if format == 'json' or request.is_ajax():
-        return json_upload_detail(request, upload)
+        with statsd.timer('devhub.upload_detail'):
```
Comment (kumar303) [inline apps/devhub/views.py]: This one is at https://app.datadoghq.com/metric/explorer?live=true&page=0&is_auto=false&from_ts=1452542782736&to_ts=1452629182736&tile_size=m&exp_metric=addons_prod.view.devhub.views.upload_detail.GET.avg&exp_scope=&exp_agg=avg&exp_row_type=metric

File: static/js/common/upload-addon.js (hunk 1 above)
Comment (kumar303) [inline static/js/common/upload-addon.js]: if a submission is quick, does this mean they'd have to wait at least 5 seconds to continue? That seems sort of long. We have enough caching that I'm not worried about hitting this URL so often.

Comment (kumar303) [inline static/js/common/upload-addon.js]: > reduce the chance we're reading while the server is writing

That shouldn't be a problem. MySQL will block the read until the transaction is committed, which is what we want. We just need to optimize the speed of committing the transaction.

**Comment by kumar303** [comment]
I think you would be a good person to add a graph for all tasks and views involved with add-on submission :) Let me know if you need any other pointers. Just add it to https://app.datadoghq.com/dash/87897/addons-apps

File: apps/devhub/tasks.py (hunk 2 above)
Comment (mstriemer) [inline apps/devhub/tasks.py]: Cool, do all tasks get timed?

**Comment by mstriemer** [comment]
I've gone to add a graph before and couldn't figure it out. I'll bug you about that in a bit.

File: apps/devhub/tasks.py (hunk 2 above)
Comment (kumar303) [inline apps/devhub/tasks.py]: yep, every task

Comment (kumar303) [inline apps/devhub/tasks.py]: ...as of last deployment (so we don't have a lot of historic data).

**Comment by kumar303** [comment]
I'd like to check these timings for load test results so I'm adding a graph right now

File: apps/devhub/views.py [hunk 7]
```diff
@@ -677,8 +679,15 @@ def standalone_upload_detail(request, uuid):
 @dev_required
 @json_view
 def upload_detail_for_addon(request, addon_id, addon, uuid):
-    upload = get_object_or_404(FileUpload, uuid=uuid)
-    return json_upload_detail(request, upload, addon_slug=addon.slug)
+    with statsd.timer('devhub.upload_detail_for_addon'):
+        try:
+            upload = get_object_or_404(FileUpload, uuid=uuid)
+            return json_upload_detail(request, upload, addon_slug=addon.slug)
+        except:
```
Comment (kumar303) [inline apps/devhub/views.py]: this should be `except Exception:` so it doesn't catch system errors.

Comment (mstriemer) [inline apps/devhub/views.py]: It just logs the exception and re-raises it so I think this should be fine.

**Pull request merged**